packages/flutter/lib/src/material/drawer_header.dart

```diff
@@ -22,27 +27,62 @@ const double _kDrawerHeaderHeight = 140.0;
 ///  * [Drawer]
 ///  * [DrawerItem]
 ///  * <https://www.google.com/design/spec/patterns/navigation-drawer.html>
-class DrawerHeader extends StatelessWidget {
+
+class DrawerHeader extends StatefulWidget {
   /// Creates a material design drawer header.
   ///
   /// Requires one of its ancestors to be a [Material] widget.
-  const DrawerHeader({ Key key, this.background, this.content }) : super(key: key);
+  DrawerHeader({
+    Key key,
+    this.decoration,
+    this.currentAccountPicture,
+    this.otherAccountsPictures,
+    this.accountName,
+    this.accountEmail,
+    this.onDetailsPressed
+  }) : super(key: key);
+
+  /// A callback that gets called when the account name/email/dropdown
+  /// section is pressed.
+  final VoidCallback onDetailsPressed;
 
-  /// A widget that extends behind the system status bar and is stacked
-  /// behind the [content] widget.
-  final Widget background;
+  /// Decoration for the main drawer header container useful for applying
+  /// backgrounds.
+  final BoxDecoration decoration;
 
-  /// A widget that's positioned below the status bar and stacked on top of the
-  /// [background] widget. Typically a view of the user's id.
-  final Widget content;
+  /// A widget placed in the upper-left corner representing the current
+  /// account picture.
+  final Widget currentAccountPicture;
+
+  /// A widget list placed in the upper-right corner representing
+  /// pictures of other accounts.
+  ///
+  /// The maximum number of drawable pictures is 3.
```

Comment:
This might be a little clearer if you said: Up to three pictures will be displayed.